lib/vanagon/project.rb

```diff
@@ -270,22 +270,75 @@ def has_provides?
       !get_provides.empty?
     end
 
-    # Collects the preinstall packaging actions for the project and it's components
-    # for the specified packaging state
+    # Collects the preinstall packaging actions for the project and its components
+    #   for the specified packaging state
     #
     # @param pkg_state [String] the package state we want to run the given scripts for.
     #   Can be one or more of 'install' or 'upgrade'
     # @return [String] string of Bourne shell compatible scriptlets to execute during the preinstall
     #   phase of packaging during the state of the system defined by pkg_state (either install or upgrade)
-    def get_preinstall_actions(pkg_state)
-      scripts = @components.map(&:preinstall_actions).flatten.compact.select { |s| s.pkg_state.include? pkg_state }.map(&:scripts)
-      if scripts.empty?
+    def get_preinstall_actions(pkg_state) # rubocop:disable Metrics/AbcSize
+      pkg_state = Array(pkg_state)
+      preinstall_scripts = Array.new
```

Comment:
`[]` is preferred to `Array.new` for... _reasons_?

Reply:
+1

Reply:
but they don't work the same
```
irb(main):001:0> foo = ['a']
=> ["a"]
irb(main):002:0> [foo]
=> [["a"]]
irb(main):003:0> Array(foo)
=> ["a"]
```

Reply:
oh, woops. ignore me. Thought this was about the type coerscion.